modules/tinymce/src/core/main/ts/api/EditorOptions.ts

```diff
@@ -187,7 +196,7 @@ const processDefaultValue = <T, U>(name: string, defaultValue: T, processor: Pro
 
 const create = (editor: Editor, initialOptions: Record<string, unknown>): Options => {
   const registry: Record<string, OptionSpec<any, any>> = {};
-  const values: Record<string, any> = {};
+  const values: Record<string, { value: any; isDefault: boolean }> = {};
```

Comment:
Is this extra `isDefault` flag really needed you could just have it get the default value from the registry in the get function as a fallback. Sure it would make two lookups but that is usually very fast. So if you unset a value the key is deleted from the `values` and then get will return the fallback value from the `registry`. Is set would then just need to check if the value is defined in values or not.
But maybe I'm not reading the code right or missing something.

Reply:
The main reason was performance related so that we don't have to keep looking up 2 objects all the time and creating multiple `Optional` instances. I'm probably over optimizing here though so I'll swap it over.

Reply:
Okay have made that change, so does that look better/more like what you were thinking?

Reply:
Right yeah hopefully we don't pull the settings object directly that often that this will be a performance issue. If we do we can always optimize that at the call site by using a variable. This change is so much cleaner and makes the data structures simpler.

Reply:
Another option would be to not use Obj.get and options at all in the `get` function and just do key lookup directly on the objects and use isUndefined calls etc to check if they exist or not. But that could be done if it becomes a hot spot.

Reply:
Yeah I did consider doing that atm but figured we were better to just see if we do get any perf issues and optimise then instead as this really shouldn't since settings aren't looked up too often.